src/main/java/mate/academy/spring/controller/DeveloperController.java

```diff
@@ -0,0 +1,102 @@
+
+@Log4j2
+@Controller
+@RequestMapping("/developer")
+public class DeveloperController {
+
+    private final DeveloperService developerService;
+
+    @Autowired
+    public DeveloperController(DeveloperService developerService) {
+        this.developerService = developerService;
+    }
+
+    @RequestMapping(method = RequestMethod.GET)
+    public String developer() {
+        log.info("controller: DEVELOPER");
+        return "developer/developer";
+    }
+
+//    @RequestMapping(value = "/index", method = RequestMethod.GET)
+//    public String index() {
+//        log.info("controller: DEVELOPER INDEX");
+//        return "/index";
+//    }
+
+    @RequestMapping(value = "/info", method = RequestMethod.GET)
+    public String developerInfo() {
+        log.info("controller: DEVELOPER INFO");
+        return "developer/info";
+    }
+
+    @RequestMapping(value = "/delete/{developerId}", method = RequestMethod.POST)
+    public String developerDelete(@PathVariable Long developerId) {
+        log.info("controller: DEVELOPER DELETE. Delete developer with id " + developerId + ".");
+        developerService.delete(developerId);
+        return "redirect:/developer/listDev";
+    }
+
+    @RequestMapping(value = "/{developerId}", method = RequestMethod.GET)
+    public String getDeveloperInfo(@PathVariable Long developerId, ModelMap view) {
+        log.info("controller: DEVELOPER INFO by id");
+        DeveloperDto developerDto = developerService.getDeveloper(developerId);
+        view.put("developer", developerDto);
+        return "developer/info";
+    }
+
+    @RequestMapping(value = "/newDev", method = RequestMethod.GET)
+    public ModelAndView newDeveloper() {
+        log.info("controller: DEVELOPER newDev");
+        return new ModelAndView("developer/createDeveloper", "developer", new Developer());
+    }
+
+    @RequestMapping(value = "/addDev", method = RequestMethod.POST)
+    public String putDeveloper(DeveloperDto devDto, ModelMap model) {
+        log.info("controller: DEVELOPER addDev");
+
+        Developer dev = new Developer();
+
+        dev.setDeveloperAge(devDto.getDeveloperAge());
+        dev.setDeveloperSalary(devDto.getDeveloperSalary());
+        dev.setDeveloperName(devDto.getDeveloperName());
+
+        Developer developer = developerService.create(dev);
+        Long id = developer.getDeveloperId();
+        DeveloperDto developerDto = developerService.getDeveloper(id);
+
+        model.put("developer", developerDto);
+        return "developer/showDev";
+    }
+
+    @RequestMapping(value = "/listDev", method = RequestMethod.GET)
+    public String listDeveloper(ModelMap model) {
+
+        log.info("controller: DEVELOPER LIST");
+
+        List<DeveloperDto> developers = developerService.getAll();
+
+        model.put("developers", developers);
+        return "developer/listDeveloper";
+    }
+
+    @PostConstruct
+    public void postConstruct() {
+        log.info("Developer controller OK!");
+    }
+}
```

Comment:
a lot of empty lines in methods, remove them, and you don't need log.info("controller: DEVELOPER LIST") here

Reply:
@DmitriyLitvin Why? I want to see that this method are working...